receiver/azureblobreceiver/blobeventhandler.go

```diff
@@ -0,0 +1,151 @@
+// Copyright OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azureblobreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureblobreceiver"
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	eventhub "github.com/Azure/azure-event-hubs-go/v3"
+	"go.uber.org/zap"
+)
+
+type BlobEventHandler interface {
+	Run(ctx context.Context) error
+	Close(ctx context.Context) error
+	SetLogsDataConsumer(logsDataConsumer LogsDataConsumer)
+	SetTracesDataConsumer(tracesDataConsumer TracesDataConsumer)
+}
+
+type AzureBlobEventHandler struct {
+	blobClient               BlobClient
+	logsDataConsumer         LogsDataConsumer
+	tracesDataConsumer       TracesDataConsumer
+	logsContainerName        string
+	tracesContainerName      string
+	eventHubSonnectionString string
+	hub                      *eventhub.Hub
+	logger                   *zap.Logger
+}
+
+const (
+	blobCreatedEventType = "Microsoft.Storage.BlobCreated"
+)
+
+func (p *AzureBlobEventHandler) Run(ctx context.Context) error {
+
+	if p.hub != nil {
+		return nil
+	}
+
+	hub, err := eventhub.NewHubFromConnectionString(p.eventHubSonnectionString)
+	if err != nil {
+		return err
+	}
+
+	p.hub = hub
+
+	runtimeInfo, err := hub.GetRuntimeInformation(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, partitionID := range runtimeInfo.PartitionIDs {
+		_, err := hub.Receive(ctx, partitionID, p.newMessageHangdler, eventhub.ReceiveWithLatestOffset())
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *AzureBlobEventHandler) newMessageHangdler(ctx context.Context, event *eventhub.Event) error {
+	p.logger.Debug(fmt.Sprintf("New event: %s", string(event.Data)))
+
+	var eventDataSlice []map[string]interface{}
+	marshalErr := json.Unmarshal(event.Data, &eventDataSlice)
+	if marshalErr != nil {
+		p.logger.Error(marshalErr.Error())
+		return marshalErr
+	}
+
+	subject := eventDataSlice[0]["subject"].(string)
+	containerName := strings.Split(strings.Split(subject, "containers/")[1], "/")[0]
+	eventType := eventDataSlice[0]["eventType"].(string)
+	blobName := strings.Split(subject, "blobs/")[1]
```

Comment:
Instead of unmarshaling to a `map[string]interface{}`, consider defining your own struct with JSON struct tags. This will make the code cleaner and avoid panics here (what if `subject` is an int?). [An example of unmarshaling to a struct is available on the Go documentation](https://pkg.go.dev/encoding/json#example-Unmarshal). You will need to add struct tags to the fields as [documented on the Marshal function](https://pkg.go.dev/encoding/json#Marshal).